docs/marketplace/tutorials/fully-functioning-rag-application.md

```diff
@@ -0,0 +1,172 @@
+
+## 1. Install the required applications
+
+We start a new project by creating two applications: the **API Documentation Aggregator** and the **AI RAG Chat**.
+
+To begin, navigate to the design section and click on _Applications_ in the sidebar. Then, select _Create new Application_.
+
+First, search for and select the **API Documentation Aggregator**. This will install the [_API Gateway_](/runtime_suite/api-gateway/10_overview.md) and the [_Swagger Aggregator_](/runtime_suite/swagger-aggregator/10_overview.md), along with several endpoints to expose the Swagger UI for viewing the APIs exposed by the configured services and retrieving the OpenAPI compliant schema.
+
+:::note
+The [API Documentation Aggregator](/runtime_suite_applications/api-documentation-aggregator/10_overview.md) is not required for the chatbot to function, but it is recommended for using the _AI RAG Template_ APIs to generate the embeddings. You can also refer to the instructions [in this tutorial](/console/tutorials/configure-marketplace-components/api-portal.mdx).
+:::
+
+Next, proceed to set up the **AI RAG Chat**. It will prompt you to create the _API Gateway_ (using the existing one with the available listener), the _AI RAG Template_, and the _AI RAG Template Chat_, along with several endpoints to expose the service APIs for sending questions and generating embeddings.
+
+![AI RAG Chat in Marketplace](../img/ai-rag-chat-in-marketplace.png)
+
+Finally, we add the [API Portal](/runtime_suite/api-portal/10_overview.md). If it is not already included in the project, click on _Microservices_ menu in the sidebar, and then select _Create new microservice_. Choose _From Marketplace_, and on the following page, search for and create a new _API Portal_.
+
+Once you have created these services, you can safely save the configuration.
+
+## 2. Configure the AI Rag Template service
+
+The next step is to configure the _AI RAG Template_ service. From the design section, navigate to the _Microservices_ page and select the service (the default name is _ai-rag-template_, but you may have changed its name). From there, click on the _Environment Variables_ tab where you need to modify the following values:
+
+- `MONGODB_CLUSTER_URI`: This is the full connection string to MongoDB.
+- `EMBEDDING_API_KEY`: An API key to use when communicating with OpenAI for generating services. It should be the same as the OpenAI API key.
+- `LLM_API_KEY`: An API key to use when communicating with OpenAI for generating services. It should also be the same as the OpenAI API key.
+
+:::note
+`EMBEDDING_API_KEY` and `LLM_API_KEY` will have the same value: they are the API key that you can generate from [OpenAI](https://platform.openai.com).
```

Comment:
can we further explain an example on how to generate such API keys?

Reply:
There are no real guides on how to create the API. Creating one from the dashboard at platform.openai.com should be pretty straightforward.

Reply:
```suggestion
`EMBEDDING_API_KEY` and `LLM_API_KEY` will have the same value: they are the API key that you can generate from [related page on the OpenAI Developer Dashboard](https://platform.openai.com/api-keys).
```